compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -203,29 +203,67 @@ static Attribute::AttrKind fromRust(LLVMRustAttribute Kind) {
   report_fatal_error("bad AttributeKind");
 }
 
+template<typename T> static inline void AddAttribute(T *t, unsigned Index, Attribute Attr) {
+#if LLVM_VERSION_LT(14, 0)
+  t->addAttribute(Index, Attr);
+#else
+  // TODO(durin42): we should probably surface the explicit functions to Rust
+  // instead of this switch statement?
+  switch (Index) {
+    case AttributeList::ReturnIndex:
+      t->addRetAttr(Attr);
+      break;
+    case AttributeList::FunctionIndex:
+      t->addFnAttr(Attr);
+      break;
+    default:
+      t->addParamAttr(Index-AttributeList::FirstArgIndex, Attr);
+  }
```

Comment:
Can't we still use addAttributeAtIndex() here? I believe these methods were just (effectively) renamed, not really removed altogether.

Reply:
We totally can, but it felt nicer to use the more-obviously-named versions of them since we already have to have LLVM14+ and LLVM13- codepaths. Also possibly more future-proof, though aeubanks indicated the *AtIndex() methods aren't on the chopping block...

Reply:
I don't think it's useful to unpack the calls in this way here -- calling more specific functions is great if you know what kind of attribute you're dealing with and can forgo attribute indexes entirely, but this is a generic API. Effectively you're reimplementing the functionality of addAttributeAtIndex().

Reply:
Sure, done.